src/OpenTK.Windowing/GraphicsLibraryFramework/GLFWProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace OpenToolkit.Windowing.GraphicsLibraryFramework
+{
+    public static class GLFWProvider
+    {
+        private static GLFW glfw = null;
+        public static GLFW GLFW
```

Comment:
This could be replaced with `Lazy<GLFW>`

Reply:
Another problem would be that this code is not Thread safe at all. Better way to initialize it would be a static ctor which also gives the benefit of better performance as the check isn't needed at all.
If one does not like static ctors. E.g. because after unloading it once it will never be initialized again(Do we even need that?) Lazy would be a possibility as well(as already mentioned by @DylanFPS). But lazy instead of having just a null check, has a cast and a null check every time after the object is initialized so at every possible access of it's Value. Ofc performance difference can be said to be negligible, but in my opinion I don't have a problem with static ctors and would take the more performant option for a little bit less readable code.
But thats just my opinion :D